arcgis-ios-sdk-samples/Scenes/Animate image overlay/AnimateImagesWithImageOverlayViewController.swift

```diff
@@ -0,0 +1,211 @@
+// Copyright 2020 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class AnimateImagesWithImageOverlayViewController: UIViewController {
+    // MARK: Storyboard views
+    
+    /// The button to toggle image overlay animation.
+    @IBOutlet weak var playPauseButton: UIBarButtonItem!
+    /// The button to choose a frame rate for the animation
+    @IBOutlet weak var frameRateButton: UIBarButtonItem!
+    /// The label to display opacity percentage.
+    @IBOutlet weak var opacityLabel: UILabel!
+    /// The slider to change opacity level, from transparent 0% to opaque 100%.
+    @IBOutlet weak var opacitySlider: UISlider! {
+        didSet {
+            opacityLabel.text = getOpacityString()
+        }
+    }
+    
+    /// The scene view managed by the view controller.
+    @IBOutlet weak var sceneView: AGSSceneView! {
+        didSet {
+            sceneView.scene = makeScene()
+            let point = AGSPoint(x: -116.621, y: 24.7773, z: 856977.0, spatialReference: .wgs84())
+            let camera = AGSCamera(location: point, heading: 353.994, pitch: 48.5495, roll: 0)
+            sceneView.setViewpointCamera(camera)
+            sceneView.imageOverlays.add(imageOverlay)
+        }
+    }
+
+    // MARK: Instance properties
+    
+    /// A circular loop counter within a length.
+    private var loopCounter: LoopCounter!
+    /// The graphics overlay to show a graphic at the tapped point.
+    let imageOverlay = AGSImageOverlay()
+    /// The frame rate in the unit of frame per second.
+    var frameRate = 60.0
+    /// An array to hold the URLs for the overlay images.
+    var imageURLs: [URL] = []
```

Comment:
During the meeting on May 1st, Trevor mentioned it might be better to preload all the images/image frames into an array in the memory for Kotlin, so that the disk/file loading performance won't lag the animation.

I tested with loading all 240 images into the memory as `UIImage`, and a rough profiling gives me ~40 more MB memory used, which is acceptable. However, loading from disk on-the-fly does not downgrade the overall performance. So I'll leave it here for further discussion.

Reply:
I've done further profiling and decided to keep an array of `UIImage`s instead.

Keeping an array of `AGSImageFrame` consume more than 2GB memory, whereas `[UIImage]` uses less than 100MB.